src/server/services/graph/timeVaryingPathConversion.js

```diff
@@ -0,0 +1,144 @@
+
+        // For each edge, find the segment covering this time range
+        for (const [edgeIndex, segments] of edgeSegments.entries()) {
+            // Try to find segment in forward direction
+            let seg = segments.find(s =>
+                parsePostgresDate(s.startTime) <= startTime &&
+                parsePostgresDate(s.endTime) >= endTime &&
+                s.sourceId === path[edgeIndex].id &&
+                s.destinationId === path[edgeIndex + 1].id
+            );
+
+            let invert = false;
+
+            if (!seg) {
+                // Try to find segment in reverse direction
+                seg = segments.find(s =>
+                    parsePostgresDate(s.startTime) <= startTime &&
+                    parsePostgresDate(s.endTime) >= endTime &&
+                    s.sourceId === path[edgeIndex + 1].id &&
+                    s.destinationId === path[edgeIndex].id
+                );
+                if (seg) {
+                    // Check bidirectional in conversions table
+                    const bidirectional = await isBidirectional(s.sourceId, s.destinationId, conn);
+                    if (bidirectional) invert = true;
+                    else seg = null; // Not bidirectional, can't use
+                }
+            }
+
+            if (!seg) {
+                valid = false;
+                break;
+            }
+
+            let segSlope = seg.slope;
+            let segIntercept = seg.intercept;
+            if (invert) {
+                [segSlope, segIntercept] = invertConversion(segSlope, segIntercept);
+            }
+
+            [slope, intercept] = updatedConversion(slope, intercept, segSlope, segIntercept);
+        }
+        if (valid) {
+            results.push({
+                source: path[0].id,
+                destination: path[path.length - 1].id,
+                start_time: toPostgresTimestamp(startTime),
+                end_time: toPostgresTimestamp(endTime),
+                slope,
+                intercept
+            });
+        }
+    }
+
+    return results;
+}
+
+/**
+ * Chains two conversions: (slope1, intercept1) and (slope2, intercept2)
+ * Returns [slope, intercept] for the combined conversion.
+ */
+function updatedConversion(origSlope, origIntercept, newSlope, newIntercept) {
+    const slope = origSlope * newSlope;
+    const intercept = newSlope * origIntercept + newIntercept;
+    return [slope, intercept];
+}
+
+function parsePostgresDate(val) {
+
+    if (val === 'infinity') {
+        return Number.POSITIVE_INFINITY;
+    }
+    if (val === '-infinity') {
+        return Number.NEGATIVE_INFINITY;
+    }
+    if (typeof val === 'string') {
+        return new Date(val).getTime();
+    }
+}
+function toPostgresTimestamp(val) {
+    if (val === Infinity) {
+        return 'infinity';
+    }
+    if (val === -Infinity) {
+        return '-infinity';
+    }
+    return new Date(val);
+}
+
+function invertConversion(slope, intercept) {
```

Comment:
This conversion existed in src/server/services/graph/pathConversion.js. Is there a reason not to use that one?